File: app/src/main/java/mydebts/android/app/MyDebtsApplication.java
package mydebts.android.app;

import android.app.Application;

import java.util.Map;

import javax.inject.Inject;

import mydebts.android.app.di.SubcomponentBuilder;

public class MyDebtsApplication extends Application {

    @Inject Map<Class<?>, SubcomponentBuilder> subcomponentBuilders;

    @Override
    public void onCreate() {
        super.onCreate();

       DaggerMyDebtsApplicationComponent.builder()
                .myDebtsApplicationModule(new MyDebtsApplicationModule(this))
                .build().inject(this);
    }

    public SubcomponentBuilder subcomponentBuilder(Class<?> keyClass) {
        return subcomponentBuilders.get(keyClass);
    }
}

File: app/src/main/java/mydebts/android/app/MyDebtsApplication.kt
package mydebts.android.app

import android.app.Activity
import android.app.Application
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.HasActivityInjector

import javax.inject.Inject

class MyDebtsApplication : Application(), HasActivityInjector {

    @Inject lateinit var activityInjector: DispatchingAndroidInjector<Activity>

    override fun onCreate() {
        super.onCreate()

        DaggerMyDebtsComponent.builder()
                .applicationContext(applicationContext)
                .build().inject(this)
    }

    override fun activityInjector(): AndroidInjector<Activity> = activityInjector
}
